leap_ec/executable_rep/cgp.py

```diff
@@ -8,15 +9,94 @@
 from leap_ec.decoder import Decoder
 from leap_ec.int_rep.initializers import create_int_vector
 from leap_ec.int_rep.ops import mutate_randint, individual_mutate_randint
-from leap_ec.segmented_rep.initializers import create_segmented_sequence
 from .executable import Executable
 
 
+##############################
+# Primitives
+##############################
+class Primitive(ABC):
+    """Abstract class that primitive functions inherit from for CGP.
+
+    You don't need to use this class to define primitive for CGP.
+    But if you do, it allows CGP to know the arity of each function—
+    which CGPDecoder can use to prune un-needed edges in the resulting
+    graph. This sometimes leads better performance or simpler graphs.
+    """
+    @property
+    @abstractmethod
+    def arity(self) -> int:
+        """How many args are used inside the __call__ function"""
+        return 0
+
+    @abstractmethod
+    def __call__(self, *args):
+        pass
+
+class FunctionPrimitive(Primitive):
```

Comment:
For future work we can probably use python inspection to compute the arity for the given function.

Reply:
(So, not an actionable insomuch as this might manifest as a TODO for much later.)